### PR TITLE
[WEB-3917] - Link whole card

### PR DESCRIPTION
**Pull request by aralovelace**
### Card Component Improvements:
* [`src/components/Homepage/BodySection/Card/FeatureCard.tsx`](diffhunk://#diff-7d8c99d08fd51839a181d0ae06a5ba4cd80654b19eadd2105b9cddcb05b65d8bL2-R11): Updated the `FeatureCard` component to make the entire card clickable by adding a `navigate` function and conditional rendering for `image` and `links`.
* [`src/components/ProductPage/BodySection/Card/FeatureCard.tsx`](diffhunk://#diff-03149d7b17dd30385ef1ff5ff9b9db8b7764730f9733f0ac81d86f18c8bbb5a7R4-R20): Similar updates to make the card clickable and conditionally render `image` and `links`.
* [`src/components/ProductPage/BodySection/Card/QuickstartCard.tsx`](diffhunk://#diff-a0b485770ab1e4a49f6590d71aaa9092ae1d93472d9c2242e162210877756320R4-R16): Made the `QuickstartCard` component clickable and added conditional rendering for `image` and `links`.
* [`src/components/ProductPage/BodySection/Card/TutorialCard.tsx`](diffhunk://#diff-354e13a19199f682ffad3ad469c39ac76997cf850b7aa7b7855726c386fc251cR4-R15): Updated the `TutorialCard` component to make it clickable and conditionally render `image` and `links`.

### Image Handling:
* [`src/components/Image.tsx`](diffhunk://#diff-7c21327ea2e33e8f9977ba8303115bd920c3bddfd0a8127ebe06fdd6d8807ccbL29-R29): Modified the `Image` component to return `null` instead of `undefined` when no image is provided, ensuring better handling of missing images. [[1]](diffhunk://#diff-7c21327ea2e33e8f9977ba8303115bd920c3bddfd0a8127ebe06fdd6d8807ccbL29-R29) [[2]](diffhunk://#diff-7c21327ea2e33e8f9977ba8303115bd920c3bddfd0a8127ebe06fdd6d8807ccbR50-R51)

### Import Cleanup:
* Removed `withPrefix` import from various pages (`src/pages/index.tsx`, `src/pages/products/channels.tsx`, `src/pages/products/chat.tsx`, `src/pages/products/livesync.tsx`) as it was not being used. [[1]](diffhunk://#diff-18e0d4553c97cfc420e938ccafb4e3a688e782fa4512ba4ceae3e2a6f24c1987L1-R1) [[2]](diffhunk://#diff-0f4c6820fe34fde1ca4ed6ad8d2a36267dddf79d13db4c4eefc2987eddc35449L1-R1) [[3]](diffhunk://#diff-41a9b8284d2f5e1e4597f1e780f254bee393348049b37b6c0f53caa11cbdf7d6L2-R2) [[4]](diffhunk://#diff-0dce4ee0f8411195db9fbf07ec8af4d0f7702a96409ea91db0f19dc3ebe7ec5aL2-R2)

### Code Refactoring:
* [`src/components/ProductPage/BodySection/BodySection.tsx`](diffhunk://#diff-4d31deeb89a6a177e1f61d86261da7dd242ac662230286a27933c35f27f05a3fL4-R5): Simplified imports by aggregating card component imports into a single line.
* [`src/components/ProductPage/BodySection/Card/index.ts`](diffhunk://#diff-f0be287b54deef59373df95d47e95af7810957d53a78aae69bf9514a64d095b8R1-R5): Added exports for various card components to facilitate simpler imports elsewhere.

These changes collectively enhance the maintainability and usability of the codebase, particularly in the rendering of card components and handling of images.

A PR description indicating the purpose of the PR.

WEB-3917

## Review

Instructions on how to review the PR. 

The pages that have cards are
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/products/channels
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/products/spaces
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/products/livesync
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/products/chat
https://ably-docs-web-3917-rspysishyym.herokuapp.com/docs/products/asset-tracking

### CODE REVIEW

File: data/yaml/page-content/asset-tracking.yaml [hunk 1]
```diff
@@ -24,13 +24,15 @@ sections:
         type: feature
         content: Use the publishing SDK on the assets you want to track, and the subscribing SDK to observe the location of the assets.
         image: 'trackable-location.svg'
+        link: '/asset-tracking/using-the-sdks'
```
Comment (jamiehenson) [inline data/yaml/page-content/asset-tracking.yaml:27]: Looking at all the usages of `link` here - it occurs to me that we probably don't need to add in a new field here, as `link` replicates `links[0].href`. 

We could cut the complexity of this a little bit by removing `link` and instead offering a helper that takes `links` as an argument and safely returns the first `href` from it. This helper should be put in a location that makes sense to be imported by all the components that use it (i.e. one level up from all of them).

File: src/components/Homepage/BodySection/Card/FeatureCard.tsx [hunk 2]
```diff
@@ -2,8 +2,8 @@ import { CardProps } from '../../HomepageContent';
 import { Links } from '../../../ProductPage/BodySection/Card/Links';
 import { Image } from '../../../Image';
 
-export const FeatureCard = ({ title, content, image, links }: CardProps) => (
-  <div className="flex justify-between flex-col">
+export const FeatureCard = ({ title, content, image, links, link }: CardProps) => (
+  <a href={link} className="pointer flex justify-between flex-col">
```
Comment (jamiehenson) [inline src/components/Homepage/BodySection/Card/FeatureCard.tsx:6]: It's worth being mindful here of `validateDOMNesting` console errors that arise as a result of this ([reference](https://www.dhiwise.com/post/mastering-validatedomnesting-best-practices)). 

In short, it breaks the HTML standards if we put certain tags inside certain other tags (with `div`s, pretty much anything goes) - in this example, `a` tags from `<Links ... />` conflict with this outer `a` tag.

We should aim to reduce these to a minimum, so worth a look at restructuring this and testing the other pages here with this in mind.

File: src/components/Image.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ export const getImageFromList = (images: ImageProps[] = [], name?: string | Imag
 
 export const Image = ({ image, src, ...attribs }: ImageComponentProps) => {
   if (!image) {
-    return;
+    return null;
```
Comment (jamiehenson) [inline src/components/Image.tsx:29]: Nice, we should always return `null` from components when not rendering content 👍

File: src/components/ProductPage/BodySection/Card/FeatureCard.tsx [hunk 4]
```diff
@@ -2,19 +2,20 @@ import { CardProps } from '../../ProductPageContent';
 import { Links } from './Links';
 import { Image } from '../../../Image';
 
-export const FeatureCard = ({ title, content, image, links }: CardProps) => (
-  <div className="border border-extra-light-grey rounded-lg bg-extra-light-grey flex justify-between hover:border-mid-grey hover:text-cool-black group">
+export const FeatureCard = ({ title, content, image, links, link }: CardProps) => (
+  <a
```
Comment (jamiehenson) [inline src/components/ProductPage/BodySection/Card/FeatureCard.tsx:6]: Similar `validateDOMNesting` concern as the other comment

**Review by jamiehenson — CHANGES_REQUESTED** [review]
Works well from what I can see in the testing pages, I just have a few comments when it comes to component and data structures. Nice cleanups around tags and returning null though 👍

**Comment by aralovelace** [comment]
thanks @jamiehenson  for your constructive feedback. They are now updated.

**Review by jamiehenson — APPROVED** [review]
Looks good, just remember to remove this before merging 👍 https://github.com/ably/docs/pull/2253/files#diff-bbe0b63a7d562937af3cc869d1be2ff929c38015592548ba94095a6ac6c6f4a8R27

**Review by m-hulbert — APPROVED** [review]
LGTM!

**Pull request merged**